### PR TITLE
Only provide a default lineHeight when the fontSize is default

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
Our custom `Text` component provided a default line height. However, for larger fonts, that caused the text to appear cut off. To fix, we only provide a default line height when the font size is the default.

### Fixed Issues
$ https://github.com/Expensify/Expensify.cash/issues/3964

### Tests / QA Steps
1. Open the app
1. Create a new bill split
1. Verify that the currency/amount is fully visible.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
image

#### Mobile Web
image

#### Desktop
image

#### iOS
image

#### Android
image

### CODE REVIEW

**Review by Dal-Papa — APPROVED** [review]

**Review by deetergp — DISMISSED** [review]
Considering the initial issue was logged on Android, it is espcially important that it be tested on Android. Disagree?

**Comment by roryabraham** [comment]
> Considering the initial issue was logged on Android, it is espcially important that it be tested on Android

Agree! I was having emulator issues before I ran out of time EOD on Friday when I created this PR, but I will work on getting the Android testing done here.

**Comment by roryabraham** [comment]
Sorry about that @deetergp, G2G now 👍

**Comment by roryabraham** [comment]
@deetergp Friendly bump because this resolves an hourly deploy blocker

**Review by thienlnam — APPROVED** [review]

**Comment by thienlnam** [comment]
Hmm looks like it still needs an approval from @deetergp (Since this is an hourly deploy blocker and Scott's comment was addressed going to see if I can unassign to merge)

**Comment by mountiny** [comment]
@thienlnam You need to Dismiss his review

**Comment by thienlnam** [comment]
@mountiny Thanks!

**Pull request merged**
